### PR TITLE
setup ui tests infrastructure

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds ui-tests dir with required setup to run jest on pupeteer, with
screenshot matching enabled

* Adds an example ui test

### CODE REVIEW

**Comment by harunurhan** [comment]
@drjova have look!

One thing to mention, ui-tests can't use `es6 imports` but other `es6` features are available. This a `node` thing and I don't think it is necessary to setup babel for only this.

TODO: local screenshot screenshot doesn't match on travis.

File: ui-tests/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  },
+  "globals": {
+    "page": true,
+    "browser": true
+  },
+  "rules": {
+    "no-underscore-dangle": "off",
+    "import/no-extraneous-dependencies": "off"
+  }
+}
```
Comment (drjova) [inline ui-tests/.eslintrc]: end of the line :P

File: run-ui-tests.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+yarn build
+cd ui-tests
+chown -R 999 .
+docker-compose up
```
Comment (david-caro) [inline run-ui-tests.sh]:
```
sudo chown -R 999 .
rc=0
docker-compose run --rm unit || rc=$?
sudo chown -R $USER .
return $?
```

**Comment by david-caro** [comment]
```
diff --git a/run-ui-tests.sh b/run-ui-tests.sh
index 08986cd..eccaf9d 100755
--- a/run-ui-tests.sh
+++ b/run-ui-tests.sh
@@ -1,7 +1,9 @@
+#!/bin/bash -xe
+
 yarn build
 cd ui-tests
 sudo chown -R 999 .
 rc=0
 docker-compose run --rm node || rc=$?
 sudo chown -R $USER .
-return $?
\ No newline at end of file
+exit $rc
```

**Pull request merged**